### PR TITLE
Fix for profile update and signup form cleanup

### PR DESCRIPTION
**Pull request by pkdash**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Try to edit your profile by uploading a profile photo and then saving the changes. If any of the required profile field data is missing you should see a message accordingly and the profile editing form still be displayed. Otherwise the profile photo upload should be successful. 
2. Bring up the hydroshare user signup page. You should find that all required fields are marked with '*".

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|
| NEW SECURITY HOTSPOTS | 0 |
| NEW CODE SMELLS | 0 |
| NEW DUPLICATED LINES DENSITY | 0.0 |
| NEW BUGS | 0 |
| NEW VULNERABILITIES | 0 |

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4331](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4331)

File: theme/forms.py [hunk 1]
```diff
@@ -332,19 +332,19 @@ class Meta:
 
     def clean_organization(self):
         data = self.cleaned_data['organization']
-        if len(data.strip()) == 0:
+        if data is None or len(data.strip()) == 0:
             raise forms.ValidationError("Organization is a required field.")
         return data
```
Comment (sblack-usu) [inline theme/forms.py:337]: Should this return data.strip()? As well as the others below?

File: theme/templates/accounts/sign_up_form.html [hunk 2]
```diff
@@ -219,7 +219,7 @@
 
     $('.tagsinput').tagsInput({
       interactive: true,
-      placeholder: "Organization(s)",
+      placeholder: "Organization(s) *",
```
Comment (sblack-usu) [inline theme/templates/accounts/sign_up_form.html:222]: What prompted this change?

Comment (pkdash) [reply theme/templates/accounts/sign_up_form.html:222]: Dave was testing the fix I made for profile pic upload. He was trying to create a new user account and noticed that some of the required fields are not marked visually marked as required. He asked me to fix that as part of this PR.

Comment (sblack-usu) [reply theme/templates/accounts/sign_up_form.html:222]: I see now, thanks.

**Review by sblack-usu — APPROVED** [review]

**Pull request merged**